src/App.tsx

```diff
@@ -1,14 +1,54 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [visiableTodos, setVisiableTodos] = useState<Todo[]>([]);
+  const [loader, setLoader] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [filter, setFilter] = useState('all');
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos);
+  }, []);
+
+  useEffect(() => {
+    setLoader(true);
+    getTodos()
+      .then(setVisiableTodos)
+      .finally(() => setLoader(false));
+  }, []);
+
+  useEffect(() => {
+    const filteredByQueryTodos = todos.filter((todo => todo.title.includes(query)));
+
+    switch (filter) {
+      case 'active':
+        setVisiableTodos(filteredByQueryTodos.filter((todo => todo.completed)));
+        break;
+
+      case 'completed':
+        setVisiableTodos(filteredByQueryTodos.filter((todo => !todo.completed)));
+        break;
+
+      case 'all':
+      default:
+        setVisiableTodos(filteredByQueryTodos);
+        break;
+    }
```

Comment:
it's better to move this code to the separate function as it will make a code cleaner